examples/middleware/examples/gas_oracle.rs

```diff
@@ -0,0 +1,95 @@
+use ethers_core::types::Chain;
+use ethers_etherscan::Client;
+use ethers_middleware::gas_oracle::{
+    BlockNative, EthGasStation, Etherchain, Etherscan, GasCategory, GasNow, GasOracle, Polygon,
+    ProviderOracle,
+};
+use ethers_providers::{Http, Provider};
+
+/// In Ethereum, the "gas" of a transaction refers to the amount of computation required to execute
+/// the transaction on the blockchain. Gas is typically measured in units of "gas," and the cost of
+/// a transaction is determined by the amount of gas it consumes.
+///
+/// A "gas oracle" is a tool or service that provides information about the current price of gas on
+/// the Ethereum network. Gas oracles are often used to help determine the appropriate amount of gas
+/// to include in a transaction, in order to ensure that it will be processed in a timely manner
+/// without running out of gas.
+///
+/// Ethers-rs includes a feature called "gas oracle middleware" that allows you to customize the
+/// behavior of the library when it comes to determining the gas cost of transactions.
+#[tokio::main]
+async fn main() {
+    blocknative().await;
+    etherchain().await;
+    eth_gas_station().await;
+    etherscan().await;
+    gas_now().await;
+    polygon().await;
+    provider_oracle().await;
+}
+
+async fn blocknative() {
+    let api_key: String = "YOUR-API-KEY".into();
+    let oracle = BlockNative::new(api_key).category(GasCategory::Fastest);
+    match oracle.fetch().await {
+        Ok(gas_price) => println!("Gas price is {gas_price:?}"),
+        Err(e) => println!("Cannot estimate gas: {e:?}"),
+    }
+}
+
+async fn etherchain() {
+    let oracle = Etherchain::new().category(GasCategory::Standard);
+    match oracle.fetch().await {
+        Ok(gas_price) => println!("Gas price is {gas_price:?}"),
+        Err(e) => println!("Cannot estimate gas: {e:?}"),
+    }
+}
+
+async fn eth_gas_station() {
+    let api_key: Option<&str> = Some("YOUR-API-KEY");
```

Comment:
Let's make these environment variables so we run them on CI to check that these work?

Reply:
Suggestion applied. Just 2 points:
1. Etherchain [URL](https://www.etherchain.org/api/gasPriceOracle) is broken (http 404). I commented out the example.
2. I didn't find an API KEY for Blocknative. Anyway it's not required to call the API.